include/groov/object.hpp

```diff
@@ -74,34 +68,40 @@ constexpr auto recursive_resolve([[maybe_unused]] P p) {
 } // namespace detail
 
 template <stdx::ct_string Name, typename T, std::size_t Msb, std::size_t Lsb,
-          named... SubFields>
+          fieldlike... SubFields>
 struct field : named_container<Name, SubFields...> {
-    using type = T;
+    using type_t = T;
+    constexpr static auto mask = ((1u << (Msb - Lsb + 1u)) - 1u) << Lsb;
```

Comment:
question: what if the register is `uint32_t` and the field is 32 bits wide?

Reply:
Then the compiler will complain... we need to do something similar to the msg fields in cib. At some point these fields and msg fields end up very similar...